app/code/core/Mage/SalesRule/Model/Rule.php

```diff
@@ -415,7 +414,6 @@ public function acquireCoupon($saveNewlyCreated = true, $saveAttemptCount = 10)
         if ($this->getCouponType() == self::COUPON_TYPE_SPECIFIC) {
             return $this->getPrimaryCoupon();
         }
-        /** @var Mage_SalesRule_Model_Coupon $coupon */
```

Comment:
Why delete this line?

Reply:
This comment is not needed. Both phpstan and phpstorm knows type of this variable.